### PR TITLE
CEDS-3348 DIS-XML mapping to Exports Declaration - group 1

### PR DESCRIPTION
**Pull request by udinobi**

### CODE REVIEW

File: app/uk/gov/hmrc/exports/services/reversemapping/DeclarationId.scala [hunk 1]
```diff
@@ -14,6 +14,6 @@
  * limitations under the License.
  */
 
-package uk.gov.hmrc.exports.services.reversemapping.declaration
+package uk.gov.hmrc.exports.services.reversemapping
 
-case class XmlParsingException(message: String) extends RuntimeException(message)
+case class DeclarationId(eori: String, mrn: String)
```
Comment (glidester) [inline app/uk/gov/hmrc/exports/services/reversemapping/DeclarationId.scala:19]: Think this name is misleading. MRN is a declarationId, eori is a userId. Should be called 'context' or something like that imo.

File: app/uk/gov/hmrc/exports/services/reversemapping/declaration/ExportsDeclarationXmlParser.scala [hunk 2]
```diff
@@ -16,49 +16,53 @@
...
         createdDateTime = Instant.now(),
         updatedDateTime = Instant.now(),
         sourceId = None,
-        `type` = DeclarationType.STANDARD,
+        `type` = declarationType,
         dispatchLocation = None,
         additionalDeclarationType = additionalDeclarationType,
-        consignmentReferences = consignmentReferences,
-        linkDucrToMucr = None,
+        consignmentReferences = consignmentReferences.map(_.copy(mrn = Some(declarationId.mrn))),
```
Comment (glidester) [inline app/uk/gov/hmrc/exports/services/reversemapping/declaration/ExportsDeclarationXmlParser.scala:64]: I don't think setting the MRN is actually part of this ticket (only `lrn` and `ducr` consignmentReferences fields are mentioned). Probably better to leave it as None.

Comment (udinobi) [reply app/uk/gov/hmrc/exports/services/reversemapping/declaration/ExportsDeclarationXmlParser.scala:64]: Problem is that the MRN is not part of any of the `XML mapping` tickets. Still I think it does not hurt to have it as reminder of what we are supposed to do if the data cannot be directly derived from parsing the xml.

Comment (udinobi) [reply app/uk/gov/hmrc/exports/services/reversemapping/declaration/ExportsDeclarationXmlParser.scala]: Actually the MRN we'll be sending to DIS is not the same one in ConsignmentReferences, so I'll leave it to None :+1:

**Review by glidester — APPROVED** [review]

**Pull request merged**